metricsmentor/urls.py

```diff
@@ -47,8 +47,9 @@
          name='calculate_pvalue'),
     path('calculate_critical/', views.calculate_critical_value,
          name='calculate_critical_value'),
-    path('api/save-sim1-graph/', views.SaveSim1GraphView.as_view(),
-         name='save_sim1_graph'),
+    re_path(r'^course/(?P<pk>\d+)/api/save-sim1-graph/',
```

Comment:
Since you're using re_path here, I think this route should have `$` at the end of it, to match to the end of the string:
```
re_path(r'^course/(?P<pk>\d+)/api/save-sim1-graph/$',
```